pkg/operator/ceph/version/version.go

```diff
@@ -37,7 +38,7 @@ var (
 )
 
 func (v *CephVersion) String() string {
-	return fmt.Sprintf("%d.%d.%d %s",
+	return fmt.Sprintf("ceph version %d.%d.%d %s",
```

Comment:
@noahdesu get this branch, remove `ceph version` from this line and you'll see the panic.
Just deploy 1mon/mgr/osd and the osd init container will fail. Thanks!

Reply:
I did not see a panic, but I saw the expected failure to parse the version string:

```
2019-04-09 18:23:17.807164 I | cephosd: updating config for osd 0
failed to write osd config file. failed to create default ceph config. failed to extract ceph version. failed to parse version from: 13.2.5 mimic
```

Reply:
Weird! I'm getting:

```
2019-04-09 20:46:56.592114 I | cephosd: updating config for osd 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x168aaac]

goroutine 1 [running]:
github.com/rook/rook/pkg/daemon/ceph/osd.writeConfigFile(0xc000516460, 0xc000508460, 0x2ef0d00, 0xc000010720, 0x1a, 0x1, 0x0)
        /home/leseb/go/src/github.com/rook/rook/pkg/daemon/ceph/osd/device.go:508 +0x48c
github.com/rook/rook/pkg/daemon/ceph/osd.WriteConfigFile(0xc000508460, 0x2ef0d00, 0xc0003c5e30, 0x0, 0x0, 0x0, 0x0, 0x240, 0x400, 0x400, ...)
        /home/leseb/go/src/github.com/rook/rook/pkg/daemon/ceph/osd/device.go:494 +0x3d8
github.com/rook/rook/cmd/rook/ceph.writeOSDConfig(0x2edefc0, 0x2f12500, 0x0, 0x0, 0x0, 0x0)
        /home/leseb/go/src/github.com/rook/rook/cmd/rook/ceph/osd.go:216 +0x730
github.com/rook/rook/vendor/github.com/spf13/cobra.(*Command).execute(0x2edefc0, 0x2f12500, 0x0, 0x0, 0x2edefc0, 0x2f12500)
        /home/leseb/go/src/github.com/rook/rook/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/rook/rook/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2ee1100, 0x2ee0060, 0x2ee0c40, 0x2edd5a0)
        /home/leseb/go/src/github.com/rook/rook/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/rook/rook/vendor/github.com/spf13/cobra.(*Command).Execute(0x2ee1100, 0x405700, 0xc000084058)
        /home/leseb/go/src/github.com/rook/rook/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /home/leseb/go/src/github.com/rook/rook/cmd/rook/main.go:33 +0x36
```

Reply:
Have you added any debugging to check for things that are nil other than versioning? For example, `cephconfig` in `writeConfig`? In the error I'm getting I see the entire call stack getting unwound based on that funky error message that has all the different levels concatenated. If you were hitting a segfault along that return path the whole error message may be missing making it look like its originating in the version.go stuff.

Reply:
No... :/ My Go version: `go version go1.11.2 linux/amd64`

Reply:
I'm not too sure how the Go version is relevant. Did you find something new? Or maybe I was not clear?

Reply:
I don't think too, that was the only thing I could think of. I stopped digging and addressed all the comments, please let me know if that's ok.